tests/integration/suite/test_notifier.py

```diff
@@ -0,0 +1,62 @@
+import os
+import subprocess
+from .common import random_str
+from .conftest import cluster_and_client, ClusterContext
+
+kube_fname = os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                          "k8s_kube_config")
+
+
+def test_notifier_smtp_password(admin_mc, remove_resource):
+    client = admin_mc.client
+    name = random_str()
+    password = random_str()
+    notifier = client.create_notifier(clusterId="local",
+                                      name=name,
+                                      smtpConfig={
+                                          "defaultRecipient": "test",
+                                          "host": "test",
+                                          "port": "587",
+                                          "sender": "test",
+                                          "tls": "true",
+                                          "username": "test",
+                                          "password": password
+                                      })
+    remove_resource(notifier)
+    assert notifier is not None
+
+    assert "password" not in notifier.smtpConfig
+
+    validate_password_exists_in_crd(admin_mc, password)
+
+    notifier = client.update(notifier, smtpConfig={"username": "test2"})
+    notifier = client.reload(notifier)
+    assert notifier["smtpConfig"]["username"] == "test2"
+
+    validate_password_exists_in_crd(admin_mc, password)
+
+
+def validate_password_exists_in_crd(admin_mc, password):
+    cluster, client = cluster_and_client('local', admin_mc.client)
+    cc = ClusterContext(admin_mc, cluster, client)
+    create_kubeconfig(cc.cluster)
```

Comment:
we don't need to create kubeconfig to check the output of password, we can use k8s client, 
for example, crdClient used here: https://github.com/rancher/rancher/blob/master/tests/integration/suite/test_password_store.py#L175